bin/v-restore-user

```diff
@@ -720,82 +721,145 @@ fi
+        if [ -z "$check_config" ]; then
+            parse_object_kv_list $(cat $tmpdir/db/$database/$backup_system/db.conf)
+            if [ -z "$MD5" ]; then
+                echo "Warning!: Database ${DB} backup does not contain a password for user ${DBUSER}."
+                echo "- Please use the web interface to set a password after the restore process finishes."
+            fi
+
+            # Adjust names by removing the old user's prefix and prefixing with the current user
+            DB=$(echo "$DB" | sed -e "s/${old_user}_//")
+            DB="${user}_${DB}"
+            DBUSER=$(echo "$DBUSER" | sed -e "s/${old_user}_//")
+            DBUSER="${user}_${DBUSER}"
+
+            # Determine the host to use for the database:
+            if [ -n "$dbhost" ]; then
+                NEW_DB_HOST="$dbhost"
+            else
+                if [ "$TYPE" = "mysql" ]; then
+                    conf_file="/usr/local/hestia/conf/mysql.conf"
+                elif [ "$TYPE" = "pgsql" ]; then
+                    conf_file="/usr/local/hestia/conf/pgsql.conf"
+                else
+                    conf_file="/usr/local/hestia/conf/mysql.conf"
+                fi
+                raw_host=$(grep "^HOST=" "$conf_file" | head -n1 | cut -d"'" -f2)
+                IFS=',' read -ra host_array <<< "$raw_host"
+                num_hosts=${#host_array[@]}
+                if [ "$num_hosts" -gt 1 ]; then
+                    index=$(($RANDOM % $num_hosts))
+                    NEW_DB_HOST=${host_array[$index]}
+                else
+                    NEW_DB_HOST=${host_array[0]}
+                fi
+            fi
+
+            # Build and save the database configuration line
+            str="DB='$DB' DBUSER='$DBUSER' MD5='$MD5' HOST='$NEW_DB_HOST' TYPE='$TYPE' CHARSET='$CHARSET' U_DISK='$U_DISK' SUSPENDED='no' TIME='$(date +%T)' DATE='$(date +%F)'"
+            echo $str >> $USER_DATA/db.conf
+        else
+            parse_object_kv_list $(grep "DB='$database'" $USER_DATA/db.conf)
+        fi
+
+        # Decompress the database dump (using zstd or gzip)
+        if [ $backup_mode = 'zstd' ]; then
+            pzstd -d $tmpdir/db/$database/$database.*.sql.zst
+        else
+            gzip -d $tmpdir/db/$database/$database.*.sql.gz
+        fi
+
+        # Define the path to the decompressed dump file
+        database_dump="$tmpdir/db/$database/$database.$TYPE.sql"
+
+        if [ "$TYPE" = "mysql" ]; then
+            # Generate the temporary MySQL defaults file with correct credentials
+            mysql_connect "$NEW_DB_HOST"
+
+            # Check if the database exists (use -N to avoid headers)
+            exists=$(mysql --defaults-extra-file="$mycnf" -N -e "SHOW DATABASES LIKE '$DB';")
+            if [ -z "$exists" ]; then
+                echo "Database $DB does not exist. Creating it..."
+                mysql --defaults-extra-file="$mycnf" -e "CREATE DATABASE \`$DB\` CHARACTER SET $CHARSET;"
```

Comment:
mysql: Deprecated program name. It will be removed in a future release, use '/usr/bin/mariadb' instead

This is every where an issue